pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -234,6 +381,7 @@ func (p *Propagator) processEvent() {
 			return
 		}
 		p.removeNode(existingNode)
+		// this doesn't require a RLock, because it's in the same thread that updates the dependents.
```

Comment:
Reworded. I was trying to say there is only one Writer, and this Read operation is in the same function as the Writer, so no need to RLock.